src/CommandingOfficers/Commander.java

```diff
@@ -384,4 +394,66 @@ public GameAction getNextAIAction(GameMap gameMap)
     }
     return null;
   }
+
+  /**
+   * Private method, same signature as in Serializable interface
+   *
+   * @param stream
+   * @throws IOException
+   */
+  private void writeObject(ObjectOutputStream stream) throws IOException
+  {
+    stream.defaultWriteObject();
+
+    // save our index into the AILibrary
+    if( null == aiController )
+      stream.writeInt(0); // Humans live at index 0 of the AI array. That sounds philosophical.
+    else
+    {
+      for( AIMaker AI : AILibrary.getAIList() )
+      {
+        if( AI.getName().equalsIgnoreCase(aiController.getAIInfo().getName()) )
+          stream.writeInt(AILibrary.getAIList().indexOf(AI));
+      }
+    }
+
+    // Write out our shopping list as a flattened 2D array of booleans; for each terrain, for all unit types, can I build it?
+    // This is probably inefficient use of space, but just look at all this not-caring I can do!
```

Comment:
I'm sure you leave comments just like at your job too, don't you?

Reply:
Not quite, but mildly close.

It occurs to me that this function will break any time we add new terrains, so I guess the idea of updating the list of terrains without breaking saves is busted unless we find a better way.